### PR TITLE
detect: log relevant frames app-layer metdata

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6973

Describe changes:
- detect: log relevant frames app-layer metdata
- frames: change rust API to make tx_id mandatory
- frames: set right tx_id for SIP and websocket

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1801

https://github.com/OISF/suricata/pull/10924 with additional commit and SV test

### CODE REVIEW

File: rust/src/dns/dns.rs [hunk 1]
```diff
@@ -437,6 +437,7 @@ impl DNSState {
             input,
             input.len() as i64,
             DnsFrameType::Pdu as u8,
+            None,
```
Comment (catenacyber) [inline rust/src/dns/dns.rs:440]: @jasonish do you want to set tx_id for dns frames ?

File: rust/src/mqtt/mqtt.rs [hunk 2]
```diff
@@ -436,6 +436,7 @@ impl MQTTState {
                         current,
                         (current.len() - rem.len()) as i64,
                         MQTTFrameType::Pdu as u8,
+                        None,
```
Comment (catenacyber) [inline rust/src/mqtt/mqtt.rs:439]: @satta should we set tx_id for MQTT frames ?

File: rust/src/rfb/rfb.rs [hunk 3]
```diff
@@ -194,6 +194,7 @@ impl RFBState {
                                 current,
                                 consumed as i64,
                                 RFBFrameType::Pdu as u8,
+                                None,
```
Comment (catenacyber) [inline rust/src/rfb/rfb.rs:197]: @satta should we set tx_id for RFB frames ?

File: rust/src/sip/sip.rs [hunk 4]
```diff
@@ -168,18 +165,26 @@ impl SIPState {
...
+                    if let Ok((_, req_line)) = sip_take_line(input) {
+                        tx.request_line = req_line;
+                    }
+                    self.transactions.push_back(tx);
                     let consumed = start.len() - rem.len();
                     start = rem;
 
                     if let Some(frame) = &self.request_frame {
                         frame.set_len(flow, consumed as i64);
+                        frame.set_tx(flow, tx_id);
```
Comment (catenacyber) [inline rust/src/sip/sip.rs:187]: cc @glongo

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.

ERROR: QA failed on SURI_TLPR1_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 138 | 143 | 103.62% |
| | SURI_TLPR1_stats_chk | | |
| .uptime | 639 | 682 | 106.73% | 

Pipeline 20360

File: rust/src/dns/dns.rs (hunk 1 above)
Comment (jasonish) [reply rust/src/dns/dns.rs:440]: Yes, and it probably makes sense for all right?  Especially if we add more frames and recommend people using them.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20363

File: rust/src/dns/dns.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/dns/dns.rs:440]: It makes sense for DNS and websocket where one suricata transaction is one PDU.

I am not so sure for RFB, SMB and others... I am not sure that every byte on the wire belongs to a transaction for instance...

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/10969

File: rust/src/mqtt/mqtt.rs (hunk 2 above)
Comment (satta) [reply rust/src/mqtt/mqtt.rs:439]: What are the implications of this? (Not sure I understand the "frame" concept good enough)... if it's just adding the transaction ID to the output, then sure, why not?

File: rust/src/rfb/rfb.rs (hunk 3 above)
Comment (satta) [reply rust/src/rfb/rfb.rs:197]: See above

File: rust/src/mqtt/mqtt.rs (hunk 2 above)
Comment (catenacyber) [reply rust/src/mqtt/mqtt.rs:439]: > What are the implications of this? (Not sure I understand the "frame" concept good enough)... if it's just adding the transaction ID to the output, then sure, why not?

Transaction ID, and more important app-layer metadata for this transaction ;-)

Comment (satta) [reply rust/src/mqtt/mqtt.rs:439]: I think I have to read https://docs.suricata.io/en/latest/devguide/extending/app-layer/app-layer-frames.html and then maybe come back with some questions ;)

Comment (catenacyber) [reply rust/src/mqtt/mqtt.rs:439]: Oh, I think this doc needs some updates...

**Pull request closed without merging**